src/ekat/util/ekat_string_utils.cpp

```diff
@@ -58,6 +59,73 @@ std::string upper_case (const std::string& s) {
   return s_up;
 }
 
+double jaro_similarity (const std::string& s1, const std::string& s2) {
```

Comment:
Can you describe briefly some typical output from this algorithm and/or give a reference to it above the function declaration? That will help folks determine where it's needed. And if you have a typical use case in mind, might want to include that here too!